Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedListItemView.swift

```diff
@@ -104,16 +132,22 @@ public struct PreplannedListItemView: View {
 
 #Preview {
     PreplannedListItemView(
-        model: PreplannedMapModel(preplannedMapArea: MockPreplannedMapArea())
+        model: PreplannedMapModel(
+            offlineMapTask: OfflineMapTask(onlineMap: Map()),
+            mapArea: MockPreplannedMapArea(),
+            directory: URL.documentsDirectory
+        )!
     )
     .padding()
 }
 
 private struct MockPreplannedMapArea: PreplannedMapAreaProtocol {
-    var packagingStatus: ArcGIS.PreplannedMapArea.PackagingStatus? = .complete
+    var id: PortalItem.ID? = PortalItem.ID("012345")
+    var packagingStatus: PreplannedMapArea.PackagingStatus? = .complete
     var title: String = "Mock Preplanned Map Area"
     var description: String = "This is the description text"
-    var thumbnail: ArcGIS.LoadableImage? = nil
+    var thumbnail: LoadableImage? = nil
     
     func retryLoad() async throws { }
+    func makeParameters(using offlineMapTask: OfflineMapTask) async throws -> DownloadPreplannedOfflineMapParameters? { nil }
```

Comment:
throwing and returning an optional is most often not the right thing. This function should probably not return an optional and should just throw or fatal error.

Reply:
I decided against incorporating this feedback since returning a non-optional value for the parameters makes the units tests fail. When the parameters are `nil`, `downloadPreplannedMapArea` returns and the model's status is maintained. If the parameters are set with the default constructor or throw an error the model's status is set to `downloadFailure`.

Reply:
The test seems dubious to me then. In a real live-running situation will ever you get `nil` returned? I don't think so, and if not then we shouldn't allow it for the sake of testing. Testing should test situations and state that occur when the app runs live.